src/Analyzers/VisualBasic/Analyzers/UseAutoProperty/VisualBasicUseAutoPropertyAnalyzer.vb

```diff
@@ -157,74 +131,32 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.UseAutoProperty
             Return GetNodeToRemove(identifier)
         End Function
 
-        Protected Overrides Function IsEligibleHeuristic(field As IFieldSymbol, propertyDeclaration As PropertyBlockSyntax, semanticModel As SemanticModel, compilation As Compilation, cancellationToken As CancellationToken) As Boolean
-            If propertyDeclaration.Accessors.Any(SyntaxKind.SetAccessorBlock) Then
-                ' If this property already has a setter, then we can definitely simplify it to an auto-prop 
-                Return True
-            End If
+        Protected Overrides Sub RegisterNonConstructorFieldWrites(
+                fieldWrites As ConcurrentDictionary(Of IFieldSymbol, ConcurrentSet(Of SyntaxNode)),
+                semanticModel As SemanticModel,
+                codeBlock As SyntaxNode,
+                cancellationToken As CancellationToken)
 
             ' the property doesn't have a setter currently. check all the types the field is 
             ' declared in.  If the field is written to outside of a constructor, then this 
             ' field Is Not eligible for replacement with an auto prop.  We'd have to make 
             ' the autoprop read/write, And that could be opening up the property widely 
             ' (in accessibility terms) in a way the user would not want.
 
-            Dim propertyDecl = semanticModel.GetDeclaredSymbol(propertyDeclaration.PropertyStatement, cancellationToken)
-            If propertyDecl.DeclaredAccessibility <> Accessibility.Private Then
-                Dim containingType = field.ContainingType
-                For Each group In containingType.DeclaringSyntaxReferences.GroupBy(Function(ref) ref.SyntaxTree)
-#Disable Warning RS1030 ' Do not invoke Compilation.GetSemanticModel() method within a diagnostic analyzer
-                    Dim groupSemanticModel = If(group.Key Is semanticModel.SyntaxTree, semanticModel, compilation.GetSemanticModel(group.Key))
-#Enable Warning RS1030 ' Do not invoke Compilation.GetSemanticModel() method within a diagnostic analyzer
```

Comment:
bad code that used to have to go back to other files to do work.  now we just collect the info while processing.